packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -38,13 +38,12 @@ import {
 	InsertContext,
 	internedSpaces,
 	IRemovalInfo,
-	ISegment,
+	IMergeSegment as ISegment,
```

Comment:
this is nice, as it prevents chrun, and will keep merges simple, this is bad as it can be really confusing which type is being worked with. i'm not sure the goodness out weighs the badness, but it makes the diff simple for now. would be open to moving all of merge tree to IMergeSegment if we think the change is good.